front/app/containers/Admin/settings/areas/AreaForm/index.tsx

```diff
@@ -1,92 +1,94 @@
 import React from 'react';
-import { isEmpty, values as getValues, every } from 'lodash-es';
 
 // i18n
 import { InjectedIntlProps } from 'react-intl';
-import { FormattedMessage, injectIntl } from 'utils/cl-intl';
+import { injectIntl } from 'utils/cl-intl';
 import messages from '../messages';
 
 // components
-import { Form, Field, InjectedFormikProps, FormikErrors } from 'formik';
-import FormikInputMultilocWithLocaleSwitcher from 'components/UI/FormikInputMultilocWithLocaleSwitcher';
-import FormikQuillMultiloc from 'components/UI/QuillEditor/FormikQuillMultiloc';
-import FormikSubmitWrapper from 'components/admin/FormikSubmitWrapper';
 import { Section, SectionField } from 'components/admin/Section';
-import Error from 'components/UI/Error';
+import { Box, Button } from '@citizenlab/cl2-component-library';
 
-// Typings
-import { Multiloc } from 'typings';
+// form
+import { useForm, FormProvider } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { object } from 'yup';
+import validateMultiloc from 'utils/yup/validateMultiloc';
+import InputMultilocWithLocaleSwitcher from 'components/HookForm/InputMultilocWithLocaleSwitcher';
+import QuillMultilocWithLocaleSwitcher from 'components/HookForm/QuillMultilocWithLocaleSwitcher';
+import Feedback from 'components/HookForm/Feedback';
 
-export interface Props {}
+// typings
+import { Multiloc } from 'typings';
+import { handleHookFormSubmissionError } from 'utils/errorUtils';
 
 export interface FormValues {
   title_multiloc: Multiloc;
   description_multiloc: Multiloc;
 }
 
-class AreaForm extends React.Component<
-  InjectedFormikProps<Props & InjectedIntlProps, FormValues>
-> {
-  public static validate = (values: FormValues): FormikErrors<FormValues> => {
-    const errors: FormikErrors<FormValues> = {};
+type Props = {
+  onSubmit: (formValues: FormValues) => void | Promise<void>;
+  defaultValues?: FormValues;
+} & InjectedIntlProps;
```

Comment:
The props are defined as a type because I want an intersection between the component props and `InjectedIntlProps`. We could do this by extending an interface also but from my perfective, this is just more readable / easier to maintain and refactor in the future. I also don't foresee this having to be extended - it's the props of the component, not a reusable type.